OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -1003,6 +1004,55 @@ internal static void UpgradeActorRules(ModData modData, int engineVersion, ref L
 					}
 				}
 
+				if (engineVersion < 20171021)
+				{
+					if (node.Key.StartsWith("Capturable", StringComparison.Ordinal) || node.Key.StartsWith("ExternalCapturable", StringComparison.Ordinal))
+					{
+						// Type renamed to Types
+						var type = node.Value.Nodes.FirstOrDefault(n => n.Key == "Type");
+						if (type != null)
+							RenameNodeKey(type, "Types");
+
+						// Allow(Allies|Neutral|Enemies) replaced with a ValidStances enum
+						var stance = Stance.Neutral | Stance.Enemy;
+						var allowAllies = node.Value.Nodes.FirstOrDefault(n => n.Key == "AllowAllies");
+						if (allowAllies != null)
+						{
+							if (FieldLoader.GetValue<bool>("AllowAllies", allowAllies.Value.Value))
+								stance |= Stance.Ally;
+							else
+								stance &= ~Stance.Ally;
+
+							node.Value.Nodes.Remove(allowAllies);
+						}
+
+						var allowNeutral = node.Value.Nodes.FirstOrDefault(n => n.Key == "AllowNeutral");
+						if (allowNeutral != null)
+						{
+							if (FieldLoader.GetValue<bool>("AllowNeutral", allowNeutral.Value.Value))
+								stance |= Stance.Neutral;
```

Comment:
Does this even make sense, as you initialized `stance` with `Stance.Neutral | Stance.Enemy` (same for "AllowEnemies")?